app/controllers/menu_items_controller.rb

```diff
@@ -1,2 +1,50 @@
 class MenuItemsController < ApplicationController
-end	
\ No newline at end of file
+  before_action :authenticate_cafe
+  before_action :find_menu_item, :find_cafe, except: [:create]
+  before_action :authorize_cafe, only: [:update]
+
+  def create
+    cafe = Cafe.find_by(id: params[:cafe_id])
+    menu_item = cafe.menu_items.build(menu_item_params)
+    if !menu_item.save
+      flash[:error] = menu_item.errors.full_messages
+    end
```

Comment:
you could make this a one liner:

flash[:error]=menu_item.errors.full_messages unless menu_item.save